remix-tests/src/compiler.js

```diff
@@ -13,13 +13,21 @@ function compileFileOrFiles (filename, isDirectory, cb) {
     'tests.sol': { content: require('../sol/tests.sol.js') },
     'remix_tests.sol': { content: require('../sol/tests.sol.js') }
   }
+  // signale.debug(sources)
 
   // TODO: for now assumes filepath dir contains all tests, later all this
   // should be replaced with remix's & browser solidity compiler code
   filepath = (isDirectory ? filename : path.dirname(filename))
 
   fs.readdirSync(filepath).forEach(file => {
-    sources[file] = {content: fs.readFileSync(path.join(filepath, file)).toString()}
+    // only process .sol files
+    if (file.split('.').pop() === 'sol') {
+      let c = fs.readFileSync(path.join(filepath, file)).toString()
+      if (file.indexOf('_test.sol') > 0) {
+        c = c.replace(/(pragma solidity \^\d+\.\d+\.\d+;)/, '$1\nimport \'remix_tests.sol\';')
```

Comment:
Would it be possible to inject the `import` only if it is not already present?

Reply:
Yeah should be possible. :+1: